### DISCUSSION TITLE
Can't exclude packages in develop mode

### DISCUSSION THREAD

**Discussion by tchittesh** [pypa/setuptools#2851] [Q&A] [answered]
Suppose we have the following file structure.
```
.
├── badmodule
│   └── __init__.py
├── goodmodule
│   └── __init__.py
└── setup.py
```

with setup.py containing
```
from setuptools import setup, find_packages

setup(
        name = 'bug',
        packages=find_packages(exclude=('badmodule')),
)
```

It seems that running ```pip install -e .``` also installs the undesired ```badmodule```. (Running ```import badmodule``` works in a Python interpreter from any directory.)

However, if we do a proper non-develop mode install, these excluded modules are actually not installed.

Any idea what's going on?

**Comment by benoit-pierre** [comment, marked as answer]
That's just how an editable install work: your project source directory is added to the Python path. And yes, it means that `setup.py` is importable too...
